src/main/java/uk/org/tombolo/importer/phe/AdultObesityImporter.java

```diff
@@ -61,65 +57,165 @@ public DatasourceSpec getDatasourceSpec(String datasourceIdString) throws Except
 
     }
 
+    // Instantiating the list that will hold our .csv rows
+    private List csvRecords;
+
     @Override
     protected void importDatasource(Datasource datasource, List<String> geographyScope, List<String> temporalScope, List<String> datasourceLocation) throws Exception {
-        // Choose the apppropriate workbook sheet
-        Workbook workbook = excelUtils.getWorkbook(
-                downloadUtils.fetchInputStream(new URL(DATASOURCE), getProvider().getLabel(), DATASOURCE_SUFFIX));
-        Sheet sheet = workbook.getSheetAt(1);
-        String year = "2014";
-
-        List<TimedValueExtractor> timedValueExtractors = new ArrayList<>();
-        RowCellExtractor subjectExtractor = new RowCellExtractor(1, CellType.STRING);
-        ConstantExtractor timestampExtractor = new ConstantExtractor(year);
 
-        SubjectType subjectType = SubjectTypeUtils.getOrCreate(AbstractONSImporter.PROVIDER,
+        // We create SubjectType object that we will use to get the appropriate geometries
+        // from OaImporter class
+        SubjectType localauthority = SubjectTypeUtils.getOrCreate(AbstractONSImporter.PROVIDER,
                 OaImporter.OaType.localAuthority.name(), OaImporter.OaType.localAuthority.datasourceSpec.getDescription());
-        for (AttributeLabel attributeLabel : AttributeLabel.values()){
-            ConstantExtractor attributeExtractor = new ConstantExtractor(attributeLabel.name());
-            RowCellExtractor valueExtractor
-                    = new RowCellExtractor(getAttributeColumnId(attributeLabel), CellType.NUMERIC);
-            timedValueExtractors.add(new TimedValueExtractor(
-                    getProvider(),
-                    subjectType,
-                    subjectExtractor,
-                    attributeExtractor,
-                    timestampExtractor,
-                    valueExtractor));
+
+        // We create an empty list that will keep our .csv values
+        List<TimedValue> timedValues = new ArrayList<TimedValue>();
+
+        CSVFormat format = CSVFormat.DEFAULT;
+        String fileLocation = getDatasourceSpec("adultObesity").getUrl();
+
+        // The code below fetches the .csv file from the URL we specified in our DatasourceSpec object
+        URL url;
```

Comment:
This piece of code is more adequate if the file is local but you are reading it as URL, and originally it was used for that reason in the code. I would suggest to use the the url object directly in the fetching method, we'll be introducing the connection error messages soon.

Reply:
Again not sure if i understand...

Reply:
Pushing the change i made for now